src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,144 @@
+package persistence.study;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+
+public class ReflectionTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReflectionTest.class);
+
+    @Test
+    @DisplayName("Car 객체 이름 가져오기")
+    void showName() {
+        Class<Car> carClass = Car.class;
+        assertThat(carClass.getName()).isEqualTo("persistence.study.Car");
+    }
+
+    @Test
+    @DisplayName("Car 객체 필드 정보 가져오기")
+    void showFields() {
+        Class<Car> carClass = Car.class;
+        Field[] declaredFields = carClass.getDeclaredFields();
+        assertThat(Arrays.stream(declaredFields).count()).isEqualTo(2);
+    }
+
+    @Test
+    @DisplayName("Car 생성자들을 가져오기")
+    void showConstructors() {
+        Class<Car> carClass = Car.class;
+        Constructor<?>[] constructors = carClass.getConstructors();
+        assertThat(Arrays.stream(constructors).count()).isEqualTo(2);
+    }
+
+    @Test
+    @DisplayName("Car 메서드들을 가져오기")
+    void showMethods() {
+        Class<Car> carClass = Car.class;
+        List<String> actual = Arrays.stream(carClass.getMethods())
+                .map(Method::getName)
+                .collect(Collectors.toList());
+        assertThat(actual).contains("testGetName", "printView");
+    }
+
+    @Test
+    @DisplayName("test로 시작하는 메서드를 실행한다")
+    void executeMethodsStartingWithTest() throws Exception {
+        Class<Car> carClass = Car.class;
+        Method[] declaredMethods = carClass.getDeclaredMethods();
+        List<Method> test = Arrays.stream(declaredMethods)
+                .filter(it -> it.getName().startsWith("test"))
+                .collect(Collectors.toList());
+        Car car = carClass.getDeclaredConstructor().newInstance();
+
+        for (Method method : test) {
+            LOGGER.debug(String.valueOf(method));
+            Object invoke = method.invoke(car);
```

Comment:
```suggestion
            method.invoke(car);
```

사용하지 않는 변수는 선언하는것이 좋다고 봐요~